profile-widget/profilewidget2.cpp

```diff
@@ -1440,12 +1449,25 @@ void ProfileWidget2::contextMenuEvent(QContextMenuEvent *event)
 	QPointF scenePos = mapToScene(mapFromGlobal(event->globalPos()));
 	qreal sec_val = timeAxis->valueAt(scenePos);
 	int seconds = (sec_val < 0.0) ? 0 : (int)sec_val;
-	if (current_dive && current_dive->cylinders.nr > 1) {
+	DiveEventItem *item = dynamic_cast<DiveEventItem *>(sceneItem);
+
+	// Add or edit Gas Change
+	if (item && event_is_gaschange(item->getEvent())) {
+		int eventTime = item->getEvent()->time.seconds;
+		QMenu *gasChange = m.addMenu(tr("Edit Gas Change"));
+		for (int i = 0; i < current_dive->cylinders.nr; i++) {
+			cylinder_t *cylinder = &current_dive->cylinders.cylinders[i];
```

Comment:
I suggest
```
        const cylinder_t = get_cylinder(current_dive, i);
```
Also I'm not sure that in the planner `current_dive` is correct - do we have to use `&displayed_dive` there? I wish we could get finally rid of displayed_dive. :-/

Reply:
Ok - I reckon that these gas_change events don't even exist in the planner, so it should be fine.

Still, I would prefer if a check for `current_dive` was added to the if above so that there is not even a theoretical chance of a null-pointer access here.